### PR TITLE
Time - Alicia

### PR DESCRIPTION
**Pull request by aecombs**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The Model is the database -- it helps us store data and keep track of it throughout the lifetime of our projects.
Describe in your own words what the Controller is doing in Rails | The Controller allows our project to have functionality -- it keeps all possible actions in one place and talks with the View and the Model to create a dynamic webpage.
Describe in your own words what the View is doing in Rails | The View is the part of Rails becomes converted into HTML and uses information it receives from the Controller to do so.
Describe an edge-case controller test you wrote | I...didn't right any edge-case tests! Oh no!
What is the purpose of using strong params? (i.e. the params method in the controller) | It allows us to reference exactly which task we are looking at by grabbing the task's id
How are Rails migrations related to Rails models? | Migrations allow us to keep track of our database and how it changes over time, allowing us to alter our database while being revert those changes in the future -- the Models are the database.
Describe one area of Rails that are still unclear on | Something that I'm still struggling with is error handling. I forgot what "head :not_found" does except that it's less elegant or something. Also, migrations are something I'm still wrapping my head around even though I understand them much better now.

### CODE REVIEW

File: app/views/tasks/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<ul class="tasks">
+  --------
+  <% @tasks.each do |task| %>
+  <% status = "Mark Completed" if !task.completed %>
```
Comment (beccaelenzil) [inline app/views/tasks/index.html.erb:4]: Good use of a conditional in your view. Consider adding some simple styling (like a strike-through) if that task is complete.

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      head :not_found
+      return
+    end
+
+    @task.destroy
+    redirect_to tasks_path
+    return
+  end
+
+  def mark_complete
```
Comment (beccaelenzil) [inline app/controllers/tasks_controller.rb:77]: Consider how making two separate methods, for instance `mark_complete` and `mark_incomplete` would make this action idempotent.

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root to: 'tasks#index'
+
+  put 'tasks/:id', to: 'tasks#mark_complete'
+  resources :tasks
```
Comment (beccaelenzil) [inline config/routes.rb:6]: nice use of resources.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <title class="title">TaskList</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
+    <header class="header"><h3>My Tasks</h3></header>
```
Comment (beccaelenzil) [inline app/views/layouts/application.html.erb:13]: Good use of layouts to create a simple nav bar and header to enhance user experience.

**Comment by beccaelenzil** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️ Nice work!
| Routes follow RESTful conventions | ✔️
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | No, but not required. Next time :)
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | ✔️
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ✔️

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements | ✔️






## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Descriptive/Readable | ✅
| Concise | ✅
| Logical/Organized | ✅

**Review by beccaelenzil — COMMENTED** [review]
Good job on Task List! It functions as expected. You hit all of the main learning goals of creating your first rails project: implementing RESTful routes (and a custom route, too!), completing all CRUD actions, and creating appropriate views. In your next project you'll get to try out strong params, partial views, and stylin' (to name a few). Keep up the hard work.
